geopandas/tests/test_extension_array.py

```diff
@@ -493,6 +493,10 @@ def test_nargsort(self):
     def test_argsort_missing_array(self):
         pass
 
+    @not_yet_implemented
+    def test_equals(self):
+        pass
```

Comment:
This is actually implemented, so you don't need to skip it?

Reply:
Ah, right. I did that because the original pandas tests fail with `TypeError: ufunc 'bitwise_or' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''` on the last line of `equals`.
But that only happens for empty arrays `assert data[:0].equals(data[:0]) is True`. (https://travis-ci.org/github/geopandas/geopandas/jobs/685304484#L2159)

`equal_values` is `array([], dtype=float64)` and `equal_na` is `array([], dtype=bool)`. 

That is probably something to be fixed in pandas, right?

Reply:
`equal_values` should be an empty bool array, but it is coming from `equal_values = self == other`, so that might actually be an issue on our end (in `GeometryArray.__eq__`)

Reply:
Yes, need to add "dtype=bool" here when constructing the resulting array:

https://github.com/geopandas/geopandas/blob/f2c599683f612d131669764c65b1f4c07aa59c4e/geopandas/array.py#L959-L961

Reply:
(BTW, for a separate issue, but we could speed this up with a vectorized function ..)

Reply:
That fixed it. Thanks.

I'll open a follow-up issue on vectorizing `_binop`.

Reply:
#1416